src/ngMock/angular-mocks.js

```diff
@@ -803,6 +803,29 @@ angular.mock.animate = angular.module('ngAnimateMock', ['ng'])
             fn();
           });
           reflowQueue = [];
+        },
+        flush: function() {
+          $rootScope.$digest();
+          var somethingFlushed = false;
+          if (reflowQueue.length) {
+            somethingFlushed = true;
+            this.triggerReflow();
+          }
+          if ($$rAF.queue.length) {
+            somethingFlushed = true;
+            $$rAF.flush();
+          }
+          if ($$asyncCallback.queue.length) {
+            somethingFlushed = true;
+            this.triggerCallbackEvents();
+          }
+          if ($browser.deferredFns.length) {
+            somethingFlushed = true;
+            this.triggerCallbackPromise();
+          }
```

Comment:
we should put this into a loop and keep on flushing stuff until everything settles